Doc/library/typing.rst

```diff
@@ -1230,6 +1232,83 @@ These are not used in annotations. They are building blocks for creating generic
     ``covariant=True`` or ``contravariant=True``.  See :pep:`484` for more
     details.  By default, type variables are invariant.
 
+.. class:: TypeVarTuple
+
+    :class:`Type variable <TypeVar>` tuple.
+
+    Usage::
+
+        Ts = TypeVarTuple('Ts')
+        Shape = TypeVarTuple('Shape')
+
+    Vanilla type variables enable parameterization with a *fixed* number of
+    types. Type variable tuples, on the other hand, allow parameterization with
+    an *arbitrary* number of types by acting like an *arbitrary* number of type
+    variables wrapped in a tuple. For example::
+
+        class Array(Generic[*Shape]):
+            pass
+
+        from typing import NewType
+        Height = NewType('Height', int)
+        Width = NewType('Width', int)
+
+        1d_array: Array[Height] = Array()  # This is fine
+
+    With one type argument, ``Array`` behaves as if ``Shape = (T,)``, where
+    ``T`` is a type variable. That means ``Array`` behaves as if we'd written
+    ``Generic[T]``. Note the use of the unpacking operator ``*`` to bring ``T``
+    out of the tuple and directly into the square brackets.
```

Comment:
This sentence is confusing, since `*` isn't next to `T` anywhere in the example.

Reply:
Good point. I've expanded this explanatory text to make this more explicit.